File: 100-same-tree/same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        def re(n, m):
            if not n and not m:
                return True
            if not n and m or n and not m or n.val != m.val:
                return False
            return re(n.left, m.left) and re(n.right, m.right)

        return re(p,q)

File: 100-same-tree/same-tree.swift
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init() { self.val = 0; self.left = nil; self.right = nil; }
 *     public init(_ val: Int) { self.val = val; self.left = nil; self.right = nil; }
 *     public init(_ val: Int, _ left: TreeNode?, _ right: TreeNode?) {
 *         self.val = val
 *         self.left = left
 *         self.right = right
 *     }
 * }
 */
class Solution {
    func isSameTree(_ p: TreeNode?, _ q: TreeNode?) -> Bool {
        return self.recursive(p, q);
    }

    func recursive(_ left: TreeNode?, _ right: TreeNode?) -> Bool{
        if let left = left , let right = right{

            if left.val != right.val{
                return false
            }
            return self.recursive(left.left, right.left) && self.recursive(left.right, right.right);
        }else if let left = left{
            return false;
        }else if let right = right{
            return false;
        }
        else{
            return true;
        }
    }
}